esileapclient/tests/functional/v1/test_policy.py

```diff
@@ -0,0 +1,132 @@
+        self.projects = PolicyTests.projects
+        self.dummy_node = DummyNode(PolicyTests.config['dummy_node_dir'],
+                                    self.projects['authorized']['id'])
+
+    @pytest.mark.negative
+    def test_create_lessee(self):
+        """ Tests that a lessee cannot create an offer or a lease.
+            Test steps:
+            1) Attempt to create an offer as a lessee.
+            2) Check that the command failed. (returned non-zero exit code)
+            3) Check that the proper error message was sent to stderr.
+            4) Attempt to create a lease as a lessee (using lease create).
+            5) Check that the command failed. (returned non-zero exit code)
+            6) Check that the proper error message was sent to stderr. """
+        e = self.assertRaises(CommandFailed,
+                              esi.offer_create,
+                              self.clients['unauthorized-lessee'],
+                              self.dummy_node.uuid,
+                              resource_type='dummy_node')
+        self.assertIn('Access was denied to esi_leap:offer:create.',
+                      e.stderr.decode())
+
+        e = self.assertRaises(CommandFailed,
+                              esi.lease_create,
+                              self.clients['unauthorized-lessee'],
+                              self.dummy_node.uuid,
+                              self.projects['unauthorized']['name'],
+                              resource_type='dummy_node')
+        self.assertIn('Access was denied to esi_leap:lease:create.',
+                      e.stderr.decode())
+
+    @pytest.mark.negative
+    def test_offer_delete_lessee(self):
+        """ Tests that a lessee cannot delete an offer.
+            Test steps:
+            1) (authorized-owner) Create an offer for an owned node for the
+                unauthorized project.
+            2) Check that lease details were returned.
+            3) (unauthorized-lessee) Attempt to delete the new offer.
+            4) Check that the command failed. (returned non-zero exit code)
+            5) Check that the proper error message was sent to stderr.
+            6) (authorized-owner) (cleanup) Delete the offer from step 1. """
+        offer = esi.offer_create(self.clients['authorized-owner'],
+                                 self.dummy_node.uuid,
+                                 lessee=self.projects['unauthorized']['name'],
+                                 resource_type='dummy_node')
+        self.assertNotEqual(offer, {})
+        self.addCleanup(esi.offer_delete,
+                        self.clients['authorized-owner'],
+                        offer['uuid'])
+
+        e = self.assertRaises(CommandFailed,
+                              esi.offer_delete,
+                              self.clients['unauthorized-lessee'],
+                              offer['uuid'])
+        self.assertIn('Access was denied to esi_leap:offer:delete.',
```

Comment:
The latest esi-leap code changes this error message (there was a patch that caught these exceptions and returned more specific exceptions where practical).

Reply:
That must be why it failed on the remote server-- will update to match the latest esi-leap code.